static/src/javascripts/lib/report-error.ts

```diff
@@ -21,16 +12,28 @@ const convertError = (err: unknown): Error => {
 	return new Error(String(err));
 };
 
+/**
+ * Report errors to Sentry with optional tags metadata.
+ * @param err - An error object or string.
+ * @param tags - Tags to assign to the event.
+ * @param shouldThrow - Flag to optionally re-throw the error (true by default). This halts
+ *  execution to ensure the error is still logged to the console via browsers' built-in logging
+ *  for uncaught exceptions. This is optional because sometimes we log errors for tracking
+ *  non-error data.
+ * @param sampleRate - A sampling rate to apply to events, used for highly frequent errors.
+ *  A value of 0 will send no events, and a value of 1 will send all events (default).
+ */
 const reportError = (
 	err: unknown,
 	tags: Record<string, string>,
 	shouldThrow = true,
 	sampleRate = 1,
 ): void => {
 	const localError: FrontendError = convertError(err);
-	raven.captureException(localError, { tags, sampleRate });
+	if (sampleRate <= Math.random()) {
```

Comment:
Raven is currently configured to send 0.8% of errors to Sentry:
https://github.com/guardian/frontend/blob/main/static/src/javascripts/lib/raven.ts#L68

Could we make the sampling rate clearer to the developer?

Reply:
I'm not sure on the best way to solve this

Reply:
Good shout. What do you think of [this](https://github.com/guardian/frontend/pull/25797/commits/ca1ad32362b43c69c0ffa711568c5db7b7d9d790)?